src/shared/modules/logger/logger-factory.ts

```diff
@@ -0,0 +1,108 @@
+import { ConfigService } from '@nestjs/config';
+import { IncomingMessage, ServerResponse } from 'http';
+import { Params } from 'nestjs-pino';
+import { GenReqId, Options } from 'pino-http';
+import { ulid } from 'ulid';
+
+const genReqId: GenReqId = (
+  req: IncomingMessage,
+  res: ServerResponse<IncomingMessage>,
+) => {
+  const headerId = req.headers['x-request-id'];
+  const id = typeof headerId === 'string' && headerId ? headerId : ulid();
+  res.setHeader('X-Request-Id', String(id));
+  return id;
+};
```

Comment:
We will using ulid only for postgres database because improve performance but their is more lazy to be created than UUID. Could you update to use UUIDs in this case  ?

import { randomUUID } from 'crypto';